clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -2154,6 +2154,32 @@ public void testCallbackAndInterceptorHandleError() {
         }
     }
 
+    @Test
+    public void shouldNotInvokeFlushInCallback() {
+        Map<String, Object> configs = new HashMap<>();
+        configs.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9000");
+        configs.put(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, false);
```

Comment:
Is this setting relevant? Having it in this test makes me think that the test would only work with it, but I do not believe that to be the case.

Reply:
This setting aims to simplify our test case. Here's my understanding—please correct me if I'm wrong:
Since idempotence is enabled by default, if we don't explicitly set it to false, we would need to prepare additional data to go through the idempotence flow. I believe this is unnecessary for this test case, so I will add a comment to clarify. WDYT?

Reply:
OK, I see. It prevents needing to do the init producer id stuff. Makes sense to me.